src/main/scala/org/antipathy/scoozie/builder/TransitionBuilder.scala

```diff
@@ -26,36 +24,37 @@ private[scoozie] object TransitionBuilder {
-        case "kill"       => NodeWithConfig(Kill(config.getString("message")), config)
-        //build placeholders for fork,join and decision nodes
-        case "fork"     => NodeWithConfig(Fork(config.getString("name"), Seq.empty), config)
-        case "join"     => NodeWithConfig(Join(config.getString("name"), End()), config)
-        case "decision" => NodeWithConfig(Decision(config.getString("name"), End(), Seq.empty), config)
-        case unknown =>
-          throw new UnknownActionException(s"action type $unknown is invalid")
-      }
+    val actions = Seq(configSeq.map(c => (getType(c), c)).map {
+      case (HoconConstants.start, config)      => NodeWithConfig(Start(), config)
+      case (HoconConstants.distcp, config)     => NodeWithConfig(DistCPAction(config, yarnConfig), config)
+      case (HoconConstants.email, config)      => NodeWithConfig(EmailAction(config), config)
+      case (HoconConstants.filesystem, config) => NodeWithConfig(FsAction(config), config)
+      case (HoconConstants.hive, config)       => NodeWithConfig(HiveAction(config, yarnConfig), config)
+      case (HoconConstants.java, config)       => NodeWithConfig(JavaAction(config, yarnConfig), config)
+      case (HoconConstants.pig, config)        => NodeWithConfig(PigAction(config, yarnConfig), config)
+      case (HoconConstants.shell, config)      => NodeWithConfig(ShellAction(config, yarnConfig), config)
+      case (HoconConstants.spark, config)      => NodeWithConfig(SparkAction(config, yarnConfig), config)
+      case (HoconConstants.sqoop, config)      => NodeWithConfig(SqoopAction(config, yarnConfig), config)
+      case (HoconConstants.ssh, config)        => NodeWithConfig(SshAction(config), config)
+      case (HoconConstants.subworflow, config) => NodeWithConfig(SubWorkflowAction(config, yarnConfig), config)
+      case (HoconConstants.end, config)        => NodeWithConfig(End(), config)
+      case (HoconConstants.kill, config)       => NodeWithConfig(Kill(config.getString(HoconConstants.message)), config)
+      //build placeholders for fork,join and decision nodes
+      case (HoconConstants.fork, config) =>
+        NodeWithConfig(Fork(config.getString(HoconConstants.name), Seq.empty), config)
+      case (HoconConstants.join, config) => NodeWithConfig(Join(config.getString(HoconConstants.name), End()), config)
+      case (HoconConstants.decision, config) =>
+        NodeWithConfig(Decision(config.getString(HoconConstants.name), End(), Seq.empty), config)
+      case (unknown, _) =>
+        throw new UnknownActionException(s"action type $unknown is invalid")
     }: _*)
 
-    val startNode = try {
-      actions.filter(_.node.action.name.equalsIgnoreCase("start")).head
-    } catch {
-      case _: NoSuchElementException =>
-        throw new ConfigurationMissingException(s"Could not find start action")
+    val startNode = Try {
+      actions.filter(_.node.action.name.equalsIgnoreCase(HoconConstants.start)).head
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of head on collections.](https://app.codacy.com/app/SimonJPegg/scoozie/pullRequest?prid=3072461)